### PR TITLE
completes query for topics

### PR DESCRIPTION
**Pull request by joe-man**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -108,6 +108,62 @@ describe("GET", () => {
...
+                    expect(typeof article.author).toBe("string")
+                    expect(typeof article.created_at).toBe("string")
+                    expect(typeof article.votes).toBe("number")
+                    expect(typeof article.article_img_url).toBe("string")
+                    expect(article.hasOwnProperty("body")).toBe(false)
+                    expect(Object.keys(article).length).toBe(7)
+                })
+            })
+        })
+        test("200 - returns all articles with paper topics", () => {
```
Comment (nc-roisin-oreilly) [inline __tests__/app.test.js:151]: Excellent edge case

File: controllers/articles.controllers.js [hunk 2]
```diff
@@ -12,8 +13,14 @@ exports.getArticleByArticleID = ((req, res, next)=> {
...
 exports.getArticles = ((req, res, next) => {
-    selectArticles()
-    .then(({rows: articles}) => {
+    const {topic} = req.query
+    const query = [selectArticles(topic)]
+    if (topic) {
+        const topicExists = checkColumnExists("topics", "slug", topic)
+        query.push(topicExists)
+    }
+    Promise.all(query)
```
Comment (nc-roisin-oreilly) [inline controllers/articles.controllers.js:22]: Nicely done

File: models/articles.models.js [hunk 3]
```diff
@@ -16,11 +16,12 @@ exports.selectArticleByID = (article_id) => {
     })
 }
 
-exports.selectArticles = () => {
-    return db.query(`
-    SELECT article_id, title, topic, author, created_at, votes, article_img_url FROM articles
-    ORDER BY created_at desc
-    `)
+exports.selectArticles = (topic) => {
+    let query = "SELECT article_id, title, topic, author, created_at, votes, article_img_url FROM articles"
```
Comment (nc-roisin-oreilly) [inline models/articles.models.js:20]: Fab

**Review by nc-roisin-oreilly — COMMENTED** [review]
Excellent tests keep it up, PR approved :)
